### PR TITLE
Enhance job service to support multiple executions recording

### PR DESCRIPTION
**Pull request by steven-zou**
- print more logs
- add execution records for periodic job
- add launch job func in job context to launch sub job

Signed-off-by: Steven Zou <szou@vmware.com>

### CODE REVIEW

**Comment by reasonerjt** [comment]
@steven-zou could you check the coverage change?

File: src/jobservice/config.yml [hunk 1]
```diff
@@ -19,13 +19,13 @@ worker_pool:
   redis_pool:
     #redis://[arbitrary_username:password@]ipaddress:port/database_index
     #or ipaddress:port[,weight,password,database_index]
-    redis_url: "redis:6379"
+    redis_url: "localhost:6379"
     namespace: "harbor_job_service"
 
 #Logger for job
 logger:
   path: "/Users/szou/tmp/job_logs"
```
Comment (clouderati) [inline src/jobservice/config.yml]: Replace `/Users/szou` with `~` for increased flexibility.

File: src/jobservice/job/impl/default_context.go [hunk 2]
```diff
@@ -0,0 +1,151 @@
+// Copyright Project Harbor Authors. All rights reserved.
```
Comment (clouderati) [inline src/jobservice/job/impl/default_context.go]: Remove "All rights reserved" to match existing headers (see [this](https://github.com/goharbor/harbor/blob/master/src/common/config/config.go) as example).

File: src/jobservice/job/impl/default_context.go [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+	checkInFunc job.CheckInFunc
+
+	// launch job
+	launchJobFunc job.LaunchJobFunc
+
+	// other required information
+	properties map[string]interface{}
+}
+
+// NewDefaultContext ...
```
Comment (clouderati) [inline src/jobservice/job/impl/default_context.go]: nit: remove `...`

File: src/jobservice/pool/redis_pool_test.go [hunk 4]
```diff
@@ -33,17 +34,18 @@ func TestRegisterJob(t *testing.T) {
 		t.Error(err)
 	}
 
+	if _, ok := wp.IsKnownJob("fake_job"); !ok {
+		t.Error("expect known job but seems failed to register job 'fake_job'")
```
Comment (clouderati) [inline src/jobservice/pool/redis_pool_test.go]: Reword: `expected known job but registering 'fake_job' appears to have failed`

**Pull request merged**